clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -2118,18 +2118,17 @@ private void close(long timeoutMs, boolean swallowException) {
         log.trace("Closing the Kafka consumer");
         AtomicReference<Throwable> firstException = new AtomicReference<>();
 
-        long closeStart = time.milliseconds();
+        Timer closeTimer = time.timer(timeoutMs);
```

Comment:
Why is the `Math.min` not done any longer?

Reply:
That's what I asked in https://github.com/apache/kafka/pull/5407#discussion_r224997352
Since the Timer will be shared across multiple `close()` methods, having it as short as one `requestTimeoutMs` may not give sufficient time for `fetcher.close()`. Do you think it's okay to use a shared timer with the minimum for both close calls?

Reply:
We could update the timer so that the min was the `requestTimeoutMs` for the second `close` too.

Reply:
In the end I think the timer for the second `close` should be the minimum of what time is left out of `timeoutMs` and requestTimeoutMs.